Uniqplace 67 create base user profile/tamar bollak

TamarBollak0077 opened the pull request:
Hi Mini!
I’ve finished my first task — it's ready for your review.

Link to Jira: https://chaimyabrov.atlassian.net/browse/UNIQPLACE-67?actionerId=5ef8cf0cd630ac0bb00708ec&sourceType=assign%3FatlOrigin%3D6a9a416a0e894815bede40c8accddc7c

TamarBollak0077 committed:
registration page with input form

TamarBollak0077 committed:
Form validations + redirect after register + avatar in home header

MiniDzaldov commented on src/pages/Register.tsx:
```diff
@@ -0,0 +1,154 @@
+  });
+
+
+  const onSubmit = async (data: any) => {
```
Replace data: any with a typed object — you already have the schema, so it's safer and clearer.
for example:
type FormData = {
  fullName: string;
  email: string;
  password: string;
};

const onSubmit = async (data: FormData) => {

MiniDzaldov commented on src/pages/Register.tsx:
```diff
@@ -0,0 +1,154 @@
+      });
+
+      if (res.data.success) {
+        const user = res.data.user; 
```
Add a guard to ensure res.data.user exists before accessing its fields.
for example:
if (res.data.success && res.data.user) {
  const user = res.data.user;
  ...
} else {
  throw new Error("User data missing in response");
}

MiniDzaldov commented on src/pages/Login.tsx:
```diff
@@ -0,0 +1,101 @@
+        localStorage.setItem('user', JSON.stringify(res.data.user));
+        navigate('/');
+      }
+    } catch (error: any) {
```
Replace any with unknown and use axios.isAxiosError() for safer error handling.
for example: 
catch (error: unknown) {
  const message =
    axios.isAxiosError(error) && error.response?.data?.message
      ? error.response.data.message
      : 'Login failed';
  ...
}

MiniDzaldov replied:
u dont see this comment

MiniDzaldov commented on src/pages/Home.tsx:
```diff
@@ -0,0 +1,53 @@
+      {userName && (
+        <div style={{ display: 'flex', alignItems: 'center', position: 'absolute', top: 10, right: 10, gap: 8 }}>
+          <h2 style={{ margin: 0 }}>
+            Hi {userName}!
```
Consider extracting the greeting UI (`Hi {userName} + Avatar`) into a separate `<UserGreeting />` component. This improves readability and reusability, especially if similar greetings are needed in other parts of the app.

TamarBollak0077 replied:
I'll do it later, once I have the complete and updated code

MiniDzaldov reviewed (changes requested):
✨Hi Tamar,
 Excellent work! The code is clean, well-structured, and easy to follow — great example of readable and maintainable React architecture.

TamarBollak0077 committed:
Update Register.tsx and Add a guard to ensure res.data.user exists before accessing its fields

TamarBollak0077 committed:
Replace data: any with a typed object: formData

The type is currently defined locally within the Register component file. Later, I will move it to a dedicated types file and import it as needed.

TamarBollak0077 committed:
Replace any with unknown and use axios.isAxiosError() for safer error handling.

TamarBollak0077 commented:
I sent you my changes

TamarBollak0077 committed:
Replace any with unknown and use axios.isAxiosError() for safer error handling.

TamarBollak0077 reviewed (commented):
עכשיו פסדר?
